### ISSUE TITLE
Investigate issues with `includes`

### ISSUE THREAD

**Issue by glacials** [glacials/friendly_uuid#16]
It seems this code is returning no record:

```rb
@race = Race.includes(entries: [:runner]).find(params[:id])
```

whereas this code is correctly returning a record:

```rb
@race = Race.find(params[:id])
```

Discovered while going through `strict_loading` audits in Rails 6.1.

**Comment by ianbayne** [comment]
I can take a look at this 👍 Regarding the above, is your setup something like the following?

```ruby
class Race
  has_many :entries
end

class Entry
  belongs_to :race
  belongs_to :runner
end

class Runner
  has_many :entries
end

race = Race.create
runner = Runner.create
entry = Entry.create(race: race, runner: runner)
```

**Comment by ianbayne** [comment]
@glacials: I can take a look at this 👍 Regarding the above, is your setup something like the following?

```ruby
class Race
  has_many :entries
end

class Entry
  belongs_to :race
  belongs_to :runner
end

class Runner
  has_many :entries
end

race = Race.create
runner = Runner.create
entry = Entry.create(race: race, runner: runner)
```

**Comment by glacials** [comment]
Yep, [exactly](https://github.com/glacials/splits-io/blob/main/app/models/entry.rb)! Thanks for volunteering to take a look at this.

**Comment by ianbayne** [comment]
Sure thing! I should have included this above, but all three models mixing in FriendlyUUID? I.e., 

```ruby
class Race
  include FriendlyUUID

  has_many :entries
end

class Entry
  include FriendlyUUID

  belongs_to :race
  belongs_to :runner
end

etc
```

**Comment by ianbayne** [comment]
@glacials, I'm not able to reproduce the error you're seeing. Can you take a look at my [WIP PR](https://github.com/glacials/friendly_uuid/pull/17/files) and let me know if there's something I'm missing about the setup required to do so? Thanks!

**Comment by ianbayne** [comment]
Re-ping for @glacials in case you missed the above 🙇

**Comment by glacials** [comment]
Sorry for awol, have not had much free time. In my example actually only Race uses FriendlyUUID, the others use standard auto-incrementing integer IDs with no gems involved.

The link I provided above is the actual code base I encountered this in, if you're still having trouble with the test and interested enough to try to repro there. Thanks for all the effort, I really appreciate it!
